### PR TITLE
Add simple select component

### PR DESCRIPTION
**Pull request by msmanoj**
## Type

- [ ] Bug fix
- [x] New feature
- [ ] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1490

## Description
- Created a simple select component that is based on the browser native select
- Added an entry to the storybook
- Updated BLAST settings to use SimpleSelect

## Deployment URL
http://simple-select.review.ensembl.org

**Storybook:**
https://ensembl.github.io/ensembl-client/storybook/index.html?path=/story/components-shared-components-simpleselect--simple-select-default-story

## Views affected
- Storybook
- BLAST settings

### Other effects

- [ ] I have checked any requirements for Google Analytics
- [ ] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

## Possible complications
- Currently, we have Select & SimpleSelect both having the same purpose. We probably want to replace Select with SimpleSelect at some point

### CODE REVIEW

**Comment by azangru** [comment]
The button doesn't always press reliably (tested the right-hand corner)


https://user-images.githubusercontent.com/6834224/155041698-4aa356f4-0ebf-4f8d-b138-a243cefbfafe.mp4

File: src/shared/components/select/SelectOptionsPanel.tsx [hunk 1]
```diff
@@ -161,14 +160,15 @@ const SelectOptionsPanel = (props: Props) => {
   }, []);
 
   useEffect(() => {
-    document.addEventListener('click', handleClickOutside);
-    document.addEventListener('touchend', handleClickOutside);
+    setTimeout(() => document.addEventListener('click', handleClickOutside), 0);
+    // document.addEventListener('click', handleClickOutside);
+    // document.addEventListener('touchend', handleClickOutside);
```
Comment (azangru) [inline src/shared/components/select/SelectOptionsPanel.tsx:165]: ✂️

File: src/shared/components/simple-select/SimpleSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+  if (optionGroups.length === 1) {
+    return (
+      <div className={styles.select}>
+        <select
+          className={styles.selectResetDefaults}
+          defaultValue=""
+          {...rest}
+        >
+          {placeholder && <option value="">{placeholder}</option>}
```
Comment (azangru) [inline src/shared/components/simple-select/SimpleSelect.tsx:74]: [This](https://jsbin.com/nipakatori/edit?html,css,js,output) works surprisingly beautifully in Chrome (not sure about other browsers)

([from SO](https://stackoverflow.com/a/64045344/3925302))

File: src/shared/components/simple-select/SimpleSelect.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
+@import 'src/styles/common';
+
```
Comment (azangru) [inline src/shared/components/simple-select/SimpleSelect.scss:2]: Is there anything in this component that we want to make tweakable via custom properties?

**Comment by msmanoj** [comment]
> The button doesn't always press reliably (tested the right-hand corner)

I've put a fix for it

File: src/shared/components/select/SelectOptionsPanel.tsx (hunk 1 above)
Comment (msmanoj) [reply src/shared/components/select/SelectOptionsPanel.tsx]: That's done

File: src/shared/components/simple-select/SimpleSelect.tsx (hunk 2 above)
Comment (msmanoj) [reply src/shared/components/simple-select/SimpleSelect.tsx]: padding: 3px 10px;

File: src/shared/components/simple-select/SimpleSelect.scss (hunk 3 above)
Comment (msmanoj) [reply src/shared/components/simple-select/SimpleSelect.scss:2]: I've made the following properties tweakable:
- min-width: var(--select-min-width, 15ch);
- max-width: var(--select-max-width, 30ch);
- background-color: var(--select-background-color, $white);

**Comment by azangru** [comment]
Been playing with the min-width.

Shouldn't we make sure that the arrow down is safely within the right padding area?

image

**Comment by msmanoj** [comment]
> Been playing with the min-width.
> 
> Shouldn't we make sure that the arrow down is safely within the right padding area?

Well spotted! I've increased the right padding

File: src/shared/components/simple-select/SimpleSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+              ))}
+            </optgroup>
+          );
+        })}
+      </select>
+    </div>
+  );
+};
+
+const SimpleSelectAdapter = (props: SimpleSelectAdapterProps) => {
```
Comment (azangru) [inline src/shared/components/simple-select/SimpleSelect.tsx:116]: Reading through the code, I don't even see the point in this adapter component. What if you passed the props directly to the Select, and if the props contain an `optionGroups` field, then you build the Select one way, whereas if it contains the `options` field, you build it another way?

BTW, you shouldn't need to write `if ((props as OptionGroupsSelectProps).optionGroups)`. Typescript should be able to infer the type of props if you write `if ('optionGroups' in props)` etc.

Comment (msmanoj) [reply src/shared/components/simple-select/SimpleSelect.tsx]: That's done

**Comment by azangru** [comment]
Clicks on the arrowhead still don't get registered. So sorry.

https://user-images.githubusercontent.com/6834224/155709483-a6137166-eaa0-4a0e-846a-e4232d818c22.mp4

**Review by azangru — APPROVED** [review]

**Pull request merged**